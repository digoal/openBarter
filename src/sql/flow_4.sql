-- \i sql/drop_model.sql
-- \i sql/model.sql

\i sql/truncate_model.sql
set search_path = ob;
-- the fisrt draft.id used will be 1
select setval('ob_tdraft_id_seq',1,false);
-- the first noeud.id used will be 5
select setval('ob_tnoeud_id_seq',4);

select ob_fadd_account('own1','q1',100);
select ob_fadd_account('own2','q2',200);
select ob_finsert_bid('own1','q1',50,100,'q2');
select ob_finsert_bid('own2','q2',100,50,'q1');
--draft 1 formed
select ob_faccept_draft(1,'own1');
select ob_faccept_draft(1,'own2');
select count(*) from ob_tnoeud; -- bids are removed after the draft is accepted, since corresponding stocks are empty

select ob_finsert_bid('own1','q1',50,100,'q2');
select ob_finsert_bid('own2','q2',100,50,'q1'); -- bid 8
--draft 2 formed
select ob_frefuse_draft(2,'own1');
-- select * from ob_trefused; -- relation (x->y) = (8->7) refused
select refused from ob_tnoeud where id=7;
-- bid 8 remains
select ob_fadd_account('own3','q2',200);
select ob_finsert_bid('own3','q2',100,50,'q1');
-- draft 3 formed
select ob_faccept_draft(3,'own1');
select ob_faccept_draft(3,'own3');

select ob_fadd_account('own4','q1',100);
select ob_finsert_bid('own3','q1',50,100,'q2'); -- bid 10
-- draft 4 formed with bid 10 and 8


select ob_fdelete_bid(10); 
select ob_fdelete_bid(8);

select ob_fadd_account('own1','qa1',200);
select ob_fadd_account('own2','qa2',200);
select ob_fadd_account('own3','qa3',200);
select ob_fadd_account('own4','qa4',200);
select ob_fadd_account('own5','qa5',200);
select ob_fadd_account('own6','qa6',200);
select ob_fadd_account('own7','qa7',200);
select ob_fadd_account('own8','qa8',200);

select ob_finsert_bid('own1','qa1',100,100,'qa2');
select ob_finsert_bid('own2','qa2',100,100,'qa3');
select ob_finsert_bid('own3','qa3',100,100,'qa4');
select ob_finsert_bid('own4','qa4',100,100,'qa5');
select ob_finsert_bid('own5','qa5',100,100,'qa6');
select ob_finsert_bid('own6','qa6',100,100,'qa7');
select ob_finsert_bid('own7','qa7',100,100,'qa8');
select ob_finsert_bid('own8','qa8',100,100,'qa1');
--draft 5 formed
select ob_faccept_draft(5,'own1');
select ob_faccept_draft(5,'own2');
select ob_faccept_draft(5,'own3');
select ob_faccept_draft(5,'own4');
select ob_faccept_draft(5,'own5');
select ob_faccept_draft(5,'own6');
select ob_faccept_draft(5,'own7');
select ob_faccept_draft(5,'own8');
--draft 5 accepted
select count(*) from ob_tnoeud; -- bids are removed after the draft is accepted, since corresponding stocks are empty 

