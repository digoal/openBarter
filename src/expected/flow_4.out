-- \i sql/drop_model.sql
-- \i sql/model.sql
\i sql/truncate_model.sql
set search_path = ob;
truncate ob_tquality cascade;select setval('ob_tquality_id_seq',1,false);
psql:sql/truncate_model.sql:3: NOTICE:  truncate cascades to table "ob_tstock"
psql:sql/truncate_model.sql:3: NOTICE:  truncate cascades to table "ob_tnoeud"
psql:sql/truncate_model.sql:3: NOTICE:  truncate cascades to table "ob_tmvt"
psql:sql/truncate_model.sql:3: NOTICE:  truncate cascades to table "ob_tcommit"
 setval 
--------
      1
(1 row)

truncate ob_towner cascade; select setval('ob_towner_id_seq',1,false);
psql:sql/truncate_model.sql:4: NOTICE:  truncate cascades to table "ob_tstock"
psql:sql/truncate_model.sql:4: NOTICE:  truncate cascades to table "ob_tcommit"
psql:sql/truncate_model.sql:4: NOTICE:  truncate cascades to table "ob_tmvt"
psql:sql/truncate_model.sql:4: NOTICE:  truncate cascades to table "ob_tnoeud"
 setval 
--------
      1
(1 row)

truncate ob_tstock cascade; select setval('ob_tstock_id_seq',1,false);
psql:sql/truncate_model.sql:5: NOTICE:  truncate cascades to table "ob_tnoeud"
psql:sql/truncate_model.sql:5: NOTICE:  truncate cascades to table "ob_tcommit"
 setval 
--------
      1
(1 row)

truncate ob_tnoeud cascade; select setval('ob_tnoeud_id_seq',1,false);
psql:sql/truncate_model.sql:6: NOTICE:  truncate cascades to table "ob_tcommit"
 setval 
--------
      1
(1 row)

-- truncate ob_trefused cascade;
truncate ob_tdraft cascade; select setval('ob_tdraft_id_seq',1,false);
psql:sql/truncate_model.sql:8: NOTICE:  truncate cascades to table "ob_tcommit"
 setval 
--------
      1
(1 row)

truncate ob_tcommit cascade; select setval('ob_tcommit_id_seq',1,false);
 setval 
--------
      1
(1 row)

truncate ob_tmvt cascade; select setval('ob_tmvt_id_seq',1,false);
 setval 
--------
      1
(1 row)

truncate ob_towner cascade; select setval('ob_towner_id_seq',1,false);
psql:sql/truncate_model.sql:11: NOTICE:  truncate cascades to table "ob_tstock"
psql:sql/truncate_model.sql:11: NOTICE:  truncate cascades to table "ob_tcommit"
psql:sql/truncate_model.sql:11: NOTICE:  truncate cascades to table "ob_tmvt"
psql:sql/truncate_model.sql:11: NOTICE:  truncate cascades to table "ob_tnoeud"
 setval 
--------
      1
(1 row)

insert into ob_towner (name) values ('market');
set search_path = ob;
-- the fisrt draft.id used will be 1
select setval('ob_tdraft_id_seq',1,false);
 setval 
--------
      1
(1 row)

-- the first noeud.id used will be 5
select setval('ob_tnoeud_id_seq',4);
 setval 
--------
      4
(1 row)

select ob_fadd_account('own1','q1',100);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own2','q2',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_finsert_bid('own1','q1',50,100,'q2');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own2','q2',100,50,'q1');
 ob_finsert_bid 
----------------
              1
(1 row)

--draft 1 formed
select ob_faccept_draft(1,'own1');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(1,'own2');
 ob_faccept_draft 
------------------
                1
(1 row)

select count(*) from ob_tnoeud; -- bids are removed after the draft is accepted, since corresponding stocks are empty
 count 
-------
     0
(1 row)

select ob_finsert_bid('own1','q1',50,100,'q2');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own2','q2',100,50,'q1'); -- bid 8
 ob_finsert_bid 
----------------
              1
(1 row)

--draft 2 formed
select ob_frefuse_draft(2,'own1');
 ob_frefuse_draft 
------------------
                0
(1 row)

-- select * from ob_trefused; -- relation (x->y) = (8->7) refused
select refused from ob_tnoeud where id=7;
 refused 
---------
 {8}
(1 row)

-- bid 8 remains
select ob_fadd_account('own3','q2',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_finsert_bid('own3','q2',100,50,'q1');
 ob_finsert_bid 
----------------
              1
(1 row)

-- draft 3 formed
select ob_faccept_draft(3,'own1');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(3,'own3');
 ob_faccept_draft 
------------------
                1
(1 row)

select ob_fadd_account('own4','q1',100);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_finsert_bid('own3','q1',50,100,'q2'); -- bid 10
 ob_finsert_bid 
----------------
              1
(1 row)

-- draft 4 formed with bid 10 and 8
select ob_fdelete_bid(10); 
 ob_fdelete_bid 
----------------
              0
(1 row)

select ob_fdelete_bid(8);
 ob_fdelete_bid 
----------------
              0
(1 row)

select ob_fadd_account('own1','qa1',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own2','qa2',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own3','qa3',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own4','qa4',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own5','qa5',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own6','qa6',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own7','qa7',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_fadd_account('own8','qa8',200);
 ob_fadd_account 
-----------------
               0
(1 row)

select ob_finsert_bid('own1','qa1',100,100,'qa2');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own2','qa2',100,100,'qa3');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own3','qa3',100,100,'qa4');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own4','qa4',100,100,'qa5');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own5','qa5',100,100,'qa6');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own6','qa6',100,100,'qa7');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own7','qa7',100,100,'qa8');
 ob_finsert_bid 
----------------
              0
(1 row)

select ob_finsert_bid('own8','qa8',100,100,'qa1');
 ob_finsert_bid 
----------------
              1
(1 row)

--draft 5 formed
select ob_faccept_draft(5,'own1');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(5,'own2');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(5,'own3');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(5,'own4');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(5,'own5');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(5,'own6');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(5,'own7');
 ob_faccept_draft 
------------------
                0
(1 row)

select ob_faccept_draft(5,'own8');
 ob_faccept_draft 
------------------
                1
(1 row)

--draft 5 accepted
select count(*) from ob_tnoeud; -- bids are removed after the draft is accepted, since corresponding stocks are empty 
 count 
-------
     0
(1 row)

